mlflow/utils/model_utils.py

```diff
@@ -428,3 +429,68 @@ def _validate_pyfunc_model_config(model_config):
             "valid file path or of type ``dict`` with string keys.",
             error_code=INVALID_PARAMETER_VALUE,
         )
+
+
+RECORD_ENV_VAR_ALLOWLIST = {
+    # api key related
+    "API_KEY",
+    "API_TOKEN",
+    # databricks auth related
+    "DATABRICKS_HOST",
+    "DATABRICKS_USERNAME",
+    "DATABRICKS_PASSWORD",
+    "DATABRICKS_TOKEN",
+    "DATABRICKS_INSECURE",
+    "DATABRICKS_CLIENT_ID",
+    "DATABRICKS_CLIENT_SECRET",
+    "_DATABRICKS_WORKSPACE_HOST",
+    "_DATABRICKS_WORKSPACE_ID",
+    # dbconnect related
+    "SPARK_REMOTE",
+    "SPARK_LOCAL_REMOTE",
+}
+
+
+@contextlib.contextmanager
+def env_var_tracker():
+    """
+    Context manager for temporarily tracking environment variables accessed.
+    It tracks environment variables accessed during the context manager's lifetime.
+    """
+    from mlflow.environment_variables import MLFLOW_RECORD_ENV_VARS_IN_MODEL_LOGGING
+
+    if MLFLOW_RECORD_ENV_VARS_IN_MODEL_LOGGING.get():
+        original_getitem = os._Environ.__getitem__
+        original_get = os._Environ.get
+        tracked_env_names = set()
+
+        def updated_get_item(self, key):
+            result = original_getitem(self, key)
+            tracked_env_names.add(key)
+            return result
+
+        def updated_get(self, key, *args, **kwargs):
+            if key in self:
+                tracked_env_names.add(key)
+            return original_get(self, key, *args, **kwargs)
+
+        def get_tracked_env_names(self):
+            nonlocal tracked_env_names
+            tracked_env_names = {
+                x
+                for x in tracked_env_names
+                if any(env_var in x for env_var in RECORD_ENV_VAR_ALLOWLIST)
+            }
+            return tracked_env_names
```

Comment:
Do we need this method? This context manager can just collect used environment variables. We can isolate the allowlist logic.

Reply:
Emm if we don't use this method how do we extract `tracked_env_names` in the context manager? Using a global variable?
I can also improve this method to allow passing the RECORD_ENV_VAR_ALLOWLIST param :)

Reply:
can we yield tracked_env_names?

Reply:
Make sense :)